### PR TITLE
fix softmax dimension bug

### LINKED ISSUE #114: discrepancy of softmax dimension in the encoding code?
As stated by the description of Encoding equation, I assume that the softmax function is to be applied to the dimension along K.
But in this [line](https://github.com/zhanghang1989/PyTorch-Encoding/blob/master/encoding/nn/encoding.py#L104) to calculate the assignment weights, Softmax is calculated with the parameter `dim=1` i.e. the dimension along N, noticing that the input tensor X is BxNxK size claimed by the [ScaleL2 function](https://github.com/zhanghang1989/PyTorch-Encoding/blob/master/encoding/functions/encoding.py#L94). Would you please check is there any confusion? @zhanghang1989

### PR DESCRIPTION
**Pull request by d-li14**
fix #114

### CODE REVIEW

**Comment by lxtGH** [comment]
@d-li14 @zhanghang1989  according to definition of Encoding the sum weight is along dim 1, not dim2 [`api`](http://hangzhang.org/PyTorch-Encoding/functions.html#scaledl2) 
which is not consistent with the paper (Your modified) 
One is feature wised, The Other is code word wised attention, I don't  why this implementation

**Comment by zhanghang1989** [comment]
Sorry about the mistake. That is a bug introduced in an early commit https://github.com/zhanghang1989/PyTorch-Encoding/commit/8425bd86d403861f4691814f9e6038da3acbc355.

**Comment by d-li14** [comment]
@lxtGH To my knowledge, Encoding is calculated with codeword-wised. Supposing the annotation for [scaledL2 function](http://hangzhang.org/PyTorch-Encoding/functions.html#scaledl2) is correct, namely output tensor is of BxNxK size, that's axis of K (dim=2) should be reduced.
